src/pgsql.c

```diff
@@ -275,15 +331,19 @@ pgsql_prepare(struct kore_pgsql *pgsql, struct http_request *req,
 
 	if (TAILQ_EMPTY(&pgsql_conn_free)) {
 		if (pgsql_conn_count >= pgsql_conn_max) {
+			pgsql->error = kore_strdup(PGSQL_ERROR_MAX_CONN);
 			pgsql_queue_add(req);
 			return (KORE_RESULT_ERROR);
 		}
 
-		if (!pgsql_conn_create(pgsql))
+		if (!pgsql_conn_create(pgsql)) {
+			pgsql->error = kore_strdup(PGSQL_ERROR_CONNECTING);
```

Comment:
pgsql_conn_create() will set its own error under pgsql->error if it fails so no need to set the error again here.